src/states/elder/mod.rs

```diff
@@ -1050,34 +1054,19 @@ impl Elder {
     fn get_targets(
         &self,
         routing_msg: &RoutingMessage,
-    ) -> Result<(Vec<PublicId>, usize), RoutingError> {
-        // TODO: even if having chain reply based on connected_state,
-        // we remove self in targets info and can do same by not
-        // chaining us to conn_peer list here?
-        let conn_peers = self.connected_peers();
+    ) -> Result<(Vec<P2pNode>, usize), RoutingError> {
+        let conn_peers: Vec<_> = self.chain.elders().map(P2pNode::name).collect();
         let (targets, dg_size) = self.chain.targets(&routing_msg.dst, &conn_peers)?;
         Ok((
             targets
                 .into_iter()
-                .filter_map(|name| self.peer_map.get_id(&name))
-                .copied()
+                .filter_map(|name| self.chain.get_p2p_node(&name))
```

Comment:
Would it make sense for chain.targets to return P2pNodes? (Might event be possible to return `&P2pNode`